lesson_10/libraries/src/loaders/xavier_cruz_loader.ts

```diff
@@ -0,0 +1,65 @@
+import csv from 'csv-parser';
+import fs from 'fs';
+import { Credit, MediaItem, Role } from '../models/index.js';
+import { Loader } from './loader.js';
+
+export class XavierCruzLoader implements Loader {
+  getLoaderName(): string {
+    return 'xaviercruz';
+  }
+
+  async loadData(): Promise<MediaItem[]> {
+    const credits = await this.loadCredits();
+    const mediaItems = await this.loadMediaItems();
+
+    console.log(
+      `Loaded ${credits.length} credits and ${mediaItems.length} media items`,
+    );
+
+    for (let i = 0; i < mediaItems.length; i++) {
+      if (i < mediaItems.length / 2) {
+        mediaItems[i].addCredit(credits[i] as Credit);
```

Comment:
The extra credit test fails when I run `LOADER_NAME=xaviercruz npm run test`.

Reply:
Ive just noticed this as well. The switch over from the interface from the class makes it fail. Will be working on understanding why its failing.

Reply:
Should be fixed